src/bundle.c

```diff
@@ -505,7 +505,7 @@ static gboolean convert_to_casync_bundle(RaucBundle *bundle, const gchar *outbun
 	g_return_val_if_fail(outbundle, FALSE);
 	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
 
-	basepath = g_strndup(outbundle, strlen(outbundle) - 6);
+	basepath = g_strndup(outbundle, strlen(outbundle) > 6 ? strlen(outbundle) - 6 : 0);
 	storepath = g_strconcat(basepath, ".castr", NULL);
```

Comment:
This can lead to non-intuitive results:

`outbundle = "bundle"` ⇒ `basepath = ""` ⇒ `storepath = ".castr"`
`outbundle = "bundle12"`  ⇒ `basepath = "bu"` ⇒ `storepath = "bu.castr"`

Reply:
So we probably want to use if (g_str_has_suffix(outbundle, ".raucb")) and only then cut of the suffix.

Reply:
yes, that sounds pretty much to what I already had in mind. Did not check if it works in the entire logic, but should be close to the way to go.

Reply:
I've now reworked it as suggested.